data/transactions/logic/opcodes.go

```diff
@@ -33,14 +33,52 @@ const rekeyingEnabledVersion = 2
 // from being used with applications. Do not edit!
 const appsEnabledVersion = 2
 
-// opSize records the length in bytes for an op that is constant-length but not length 1
-type opSize struct {
-	cost      int
-	size      int
-	checkFunc opCheckFunc
+// opDetails records details such as non-standard costs, immediate
+// arguments, or dynamic layout controlled by a check function.
+type opDetails struct {
+	Cost       int
+	Size       int
+	checkFunc  opCheckFunc
+	Immediates []immediate
```

Comment:
consolodate with doc.go `opcodeImmediateNoteList`, either move those into opDetails.Immediates or move the new opDetails.Immediates data into opcodeImmediateNoteList